### ISSUE TITLE
Issue with dlls not unloading on Windows

### ISSUE THREAD

**Issue by Cyberboss** [tgstation/tgstation-server#1132]
Tracking this bug from Cit. 
tarot cardYesterday at 5:45 PM
@Cyberboss so uhhh I think I might have found the issue
One way or another the .dlls are targetting the .. old path
I'll test again after this current round by just hard restarting the process to see if it fixes it but I was able to make it break last night
the one starting with 4 is the directory of Live when it worked. Then, I restarted TGS and then recompiled, and it now goes to the one starting with 1. The server's process hasn't been restarted since then, and file writes with dills are completely broken.
I suspect if I hard restart the process it'll fix itself, so I'll be testing that after this round

https://cdn.discordapp.com/attachments/418907500876791808/764966959775219750/unknown.png

**Comment by silicons** [comment]
We have our rustg dll in static files
1. Recompile etc etc make sure there's 3 directories in Game, live, and A and B, live is pointing at let's say A
2. Reboot TGS, recompile again, and due to how it works live is now pointing at B
3. restart the server without restarting the process (so byond Restart()) 
4. our logging/anything involving files is now broken because while the byond working directory is actually B, the dlls are still  operating from A. since TGS tries to delete the files in A it only fails when it gets to the in-use .dll, resulting in most of the directory gone minus a few files like the rsc/dme/dmb/dlls that got interrupted by the file being in use
5. there's obviously no more data folder symlink in the old folder. the dlls still go to that. so they can't access the path specified because.. it doesn't exist. 
6. to fix this, I ended up just restarting the server process entirely, which seems to fix it by well, obviously, making a new process at which point the dlls are pointed to the right folder

I've only noticed it breaking this way when we restart TGS which is weird.
